src/plugins/intel_cpu/src/nodes/fullyconnected.cpp

```diff
@@ -115,7 +115,7 @@ bool FullyConnected::isSupportedOperation(const std::shared_ptr<const ov::Node>&
             errorMessage = "Doesn't support 'data' input with rank: " + std::to_string(inRank);
             return false;
         }
-        if ((one_of(inRank, 2u, 3u) && weightRank != 2) || (inRank == 4 && weightRank != 4)) {
+        if (one_of(inRank, 2u, 3u, 4u) && weightRank != 2) {
```

Comment:
Minor: It seems like after your changes we can natively support ND case  (N != 1) for data input. Maybe we can relax the condition regarding to `inRank` and remove `rank_a` limitation in [ConvertMatMulToFullyConnected transformation](https://github.com/openvinotoolkit/openvino/blob/b510b40542b474d20d9e93384b61d08ded8f2e8f/src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/convert_matmul_to_fc.cpp#L57)?

Reply:
Extend to ND case.